src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -3362,24 +3431,71 @@ private TypeSymbolWithAnnotations VisitOptionalImplicitConversion(BoundExpressio
 
         public override BoundNode VisitTupleLiteral(BoundTupleLiteral node)
         {
-            VisitTupleExpression(node);
+            VisitTupleExpression(node, node.ArgumentNamesOpt);
             return null;
         }
 
         public override BoundNode VisitConvertedTupleLiteral(BoundConvertedTupleLiteral node)
         {
-            VisitTupleExpression(node);
+            VisitTupleExpression(node, default);
             return null;
         }
 
-        private void VisitTupleExpression(BoundTupleExpression node)
+        private void VisitTupleExpression(BoundTupleExpression node, ImmutableArray<string> argumentNamesOpt)
         {
             var arguments = node.Arguments;
             ImmutableArray<TypeSymbolWithAnnotations> elementTypes = arguments.SelectAsArray((a, w) => w.VisitRvalueWithResult(a), this);
             var tupleOpt = (TupleTypeSymbol)node.Type;
-            _resultType = (tupleOpt is null) ?
-                default :
-                TypeSymbolWithAnnotations.Create(tupleOpt.WithElementTypes(elementTypes), NullableAnnotation.NotNullable);
+            if (tupleOpt is null)
+            {
+                _resultType = default;
```

Comment:
>default [](start = 30, length = 7)

I understand that the typeless tuple is an error case, but it raises a similar question to `null!`. Can we have a typeless but non-nullable expression? (Currently, it is typeless and oblivious)